examples/unimp_arxiv.py

```diff
@@ -0,0 +1,170 @@
+import logging
+import os
+
+import torch
+from ogb.nodeproppred import PygNodePropPredDataset
+
+import torch_geometric.transforms as T
+import torch_geometric.utils.mask as mask_util
+from torch_geometric.nn import TransformerConv
+
+logger = logging.getLogger(__file__)
+logging.basicConfig(level=logging.INFO)
+
+# Get dataset
+root = os.path.join("..", "data", "OGB")
+dataset = PygNodePropPredDataset(
+    "ogbn-arxiv",
+    root,
+    transform=T.Compose([T.ToUndirected()]),
+)
+data = dataset[0]
+num_classes = 40
+
+# Form masks for labels: mask gives labels being predicted
+split_idx = dataset.get_idx_split()
+train_mask = mask_util.index_to_mask(split_idx["train"], size=data.num_nodes)
+valid_mask = mask_util.index_to_mask(split_idx["valid"], size=data.num_nodes)
+test_mask = mask_util.index_to_mask(split_idx["test"], size=data.num_nodes)
+
+# Model parameters
+inner_dim = 16
+heads = 2
+
+
+def ratio_mask(mask: torch.Tensor, ratio: float, shuffle: bool = False):
+    r"""Modifies the existing :obj:`mask` by additioning setting :obj:`ratio`
+    of the :obj:`True` entries to :obj:`False`. Does not operate inplace.
+
+    If shuffle is required the masking proportion is not exact.
+
+    Args:
+        mask (torch.Tensor): The mask to re-mask.
+        ratio (float): The ratio of entries to remove.
+        shuffle (bool): Whether or not the mask is pre-shuffled, if so there is
+            no need to randomize which entires are set to :obj:`False`.
+    """
+    n = int(mask.count_nonzero().item())
+    new_mask = torch.ones(len(mask), dtype=torch.bool)
+    if not shuffle:
+        new_mask[mask] = (torch.rand(n) < ratio).type(torch.bool)
+    else:
+        new_mask[mask] = (torch.arange(0, n) < int(ratio * n)).type(torch.bool)
+    return mask & new_mask
+
+
+class MaskLabel(torch.nn.Module):
+    r"""A label embedding layer that replicates the label masking from `"Masked
+    Label Prediction: Unified Message Passing Model for Semi-Supervised
+    Classification" <https://arxiv.org/abs/2009.03509>`_ paper.
+
+    In the forward pass both the labels, and a mask corresponding to which
+    labels should be kept is provided. All entires that are not true in the
+    mask are returned as zero.
+
+    Args:
+        num_classes (int): Size of the number of classes for the labels
+        out_channels (int): Size of each output sample.
+    """
+    def __init__(self, num_classes, out_channels):
+        super().__init__()
+
+        self.emb = torch.nn.Embedding(num_classes, out_channels)
+        self.out_channels = out_channels
+
+    def forward(self, y: torch.Tensor, mask: torch.Tensor):
+        out = torch.zeros(y.shape[0], self.out_channels, dtype=torch.float)
```

Comment:
I think we should expect both `x` and `y` as input arguments (just like done previously). This lets us avoid the intermediate `out` creation and saves some memory.

Reply:
Hmm, can, but this means we need to do a clone.. the zero avoid this need because then we can write the non-in-place
```
x = x + label_embedding(y)
```

So we'd have to clone x here. What do you think is better?